workbench-core/datasets/src/dataSetService.ts

```diff
@@ -606,4 +581,76 @@ export class DataSetService {
       endpointId
     };
   }
+
+  private async _addDataSetExternalEndpoint(
+    request: AddDataSetExternalEndpointBaseRequest
+  ): Promise<AddDataSetExternalEndpointResponse> {
+    const {
+      authenticatedUser,
+      dataSetId,
+      subject,
+      externalEndpointName,
+      storageProvider,
+      externalRoleName,
+      kmsKeyArn,
+      vpcId
+    } = request;
+
+    const { data: permissionsData } = await this._authzPlugin.getAccessPermissions({
+      dataSetId,
+      subject
+    });
+    if (!permissionsData.permissions.length) {
+      throw new NotAuthorizedError(
+        `Subject "${subject}" does not have permission to access dataset "${dataSetId}.`
```

Comment:
I think there is a mismatch in concepts here; subject here would be 'dataset' or 'datasetPermission', not the user/group.

Reply:
I was following the inputs for `getDatasetPermissions` here: https://github.com/aws-solutions/solution-spark-on-aws/blob/develop/workbench-core/datasets/src/models/getAccessPermissionRequest.ts

Reply:
The inputs have changed to use `identity` over `subject` so I think we're back on the same page in terms of concepts